src/vmm/src/vstate/vcpu.rs

```diff
@@ -907,11 +908,15 @@ impl Vcpu {
                             .send(VcpuResponse::RestoredState)
                             .expect("vcpu channel unexpectedly closed");
                     })
-                    .map_err(|e| self.response_sender.send(VcpuResponse::Error(e)))
-                    .expect("vcpu channel unexpectedly closed");
+                    .unwrap_or_else(|e| {
+                        self.response_sender
+                            .send(VcpuResponse::Error(e))
+                            .expect("vcpu channel unexpectedly closed")
+                    });
```

Comment:
We have the same issue with the `expect()` in the wrong place on the `save_state()` path as well. I missed that during my original review 😞

Reply:
I solved it in this PR https://github.com/firecracker-microvm/firecracker/pull/2081/files#diff-ddcbd51d63a6e6145b8feb4a56c27d0aR350. Take a look!